common/lib/xmodule/xmodule/modulestore/mongo/draft.py

```diff
@@ -793,19 +793,26 @@ def delete_draft_only(root_location):
             versions_found = self.collection.find(
                 query, {'_id': True, 'definition.children': True}, sort=[SORT_REVISION_FAVOR_DRAFT]
             )
+            total_versions_found = versions_found.count()
+            published_version = [
+                version
+                for version in versions_found
+                if version.get('_id').get('revision') != MongoRevisionKey.draft
+            ][0]
```

Comment:
In the old version of the code, an empty `versions_found` would exit the `delete_draft_only` method - this change would cause an error instead, which seems undesirable.